test/unit/auth/auth.spec.ts

```diff
@@ -3052,6 +3053,506 @@ AUTH_CONFIGS.forEach((testConfig) => {
+          return (malformedAccessTokenAuth as Auth).getTenant(tenantId)
+            .should.eventually.be.rejected.and.have.property('code', 'app/invalid-credential');
+        });
+
+        it('should be rejected given an app which fails to generate access tokens', () => {
+          return (rejectedPromiseAccessTokenAuth as Auth).getTenant(tenantId)
+            .should.eventually.be.rejected.and.have.property('code', 'app/invalid-credential');
+        });
+
+        it('should resolve with a Tenant on success', () => {
+          // Stub getTenant to return expected result.
+          const stub = sinon.stub(testConfig.RequestHandler.prototype, 'getTenant')
+            .returns(Promise.resolve(serverResponse));
+          stubs.push(stub);
+          return (auth as Auth).getTenant(tenantId)
+            .then((result) => {
+              // Confirm underlying API called with expected parameters.
+              expect(stub).to.have.been.calledOnce.and.calledWith(tenantId);
+              // Confirm expected tenant returned.
+              expect(result).to.deep.equal(expectedTenant);
+            });
+        });
+
+        it('should throw an error when the backend returns an error', () => {
+          // Stub getTenant to throw a backend error.
+          const stub = sinon.stub(testConfig.RequestHandler.prototype, 'getTenant')
+            .returns(Promise.reject(expectedError));
+          stubs.push(stub);
+          return (auth as Auth).getTenant(tenantId)
+            .then((userRecord) => {
+              throw new Error('Unexpected success');
+            }, (error) => {
+              // Confirm underlying API called with expected parameters.
+              expect(stub).to.have.been.calledOnce.and.calledWith(tenantId);
+              // Confirm expected error returned.
+              expect(error).to.equal(expectedError);
+            });
+        });
+      });
+
+      describe('listTenants()', () => {
+        const expectedError = new FirebaseAuthError(AuthClientErrorCode.INTERNAL_ERROR);
+        const pageToken = 'PAGE_TOKEN';
+        const maxResult = 500;
+        const listTenantsResponse: any = {
+          tenants : [
+            {name: 'projects/project_id/tenants/tenant_id1'},
+            {name: 'projects/project_id/tenants/tenant_id2'},
+          ],
+          nextPageToken: 'NEXT_PAGE_TOKEN',
+        };
+        const expectedResult: ListTenantsResult = {
+          tenants: [
+            new Tenant({name: 'projects/project_id/tenants/tenant_id1'}),
+            new Tenant({name: 'projects/project_id/tenants/tenant_id2'}),
+          ],
+          pageToken: 'NEXT_PAGE_TOKEN',
+        };
+        const emptyListTenantsResponse: any = {
+          tenants: [],
+        };
+        const emptyExpectedResult: any = {
+          tenants: [],
+        };
+        // Stubs used to simulate underlying API calls.
+        let stubs: sinon.SinonStub[] = [];
+        afterEach(() => {
```

Comment:
Nit: Add an empty line before this for clarity.

Reply:
done